src/pages/Root/components/AppHeader/WalletMenu/WalletMenu.js

```diff
@@ -0,0 +1,233 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import styled from 'styled-components';
+import { useTranslation } from 'react-i18next';
+import get from 'lodash/get';
+import Tooltip from '@material-ui/core/Tooltip';
+
+import { resetWalletReducer, getCurrentWalletAddress } from 'src/ducks/wallet/walletDetails';
+import {
+	getTotalSynthsBalanceUSD,
+	getIsLoadedWalletBalances,
+	getSynthsWalletBalances,
+	getIsFetchingWalletBalances,
+} from 'src/ducks/wallet/walletBalances';
+import { showWalletPopup } from 'src/ducks/ui';
+import { getAvailableSynthsMap } from 'src/ducks/synths';
+
+import { ROUTES } from 'src/constants/routes';
+import { FIAT_CURRENCY_MAP, SYNTHS_MAP } from 'src/constants/currency';
+import { EMPTY_VALUE } from 'src/constants/placeholder';
+
+import {
+	shortenAddress,
+	formatCurrencyWithSign,
+	formatCurrency,
+	LONG_CRYPTO_CURRENCY_DECIMALS,
+	SHORT_CRYPTO_CURRENCY_DECIMALS,
+} from 'src/utils/formatters';
+
+import Card from 'src/components/Card';
+import { ButtonPrimaryExtraSmall, ButtonPrimarySmall } from 'src/components/Button/ButtonPrimary';
+import { HeadingSmall, DataMedium } from 'src/components/Typography';
+import { InfoBoxLabel, FlexDivCentered, TableNoResults } from 'src/shared/commonStyles';
+import Link from 'src/components/Link';
+import Table from 'src/components/Table';
+import { TABLE_PALETTE } from 'src/components/Table/constants';
+import Currency from 'src/components/Currency';
+
+const WalletMenu = ({
+	resetWalletReducer,
+	toggleDropdown,
+	currentWallet,
+	showWalletPopup,
+	totalSynthsBalanceUSD,
+	isLoadedWalletBalances,
+	synthsMap,
+	synthsWalletBalances,
+	isFetchingWalletBalances,
+}) => {
+	const { t } = useTranslation();
+
+	const logout = () => {
+		resetWalletReducer();
```

Comment:
perhaps for the future, I think it's better to actually have an action called "logout", that would do this wallet reset... we might call it from different places, and we might need to reset other stuff as well.

Reply:
yeah good call